Anime-App/Feature/Anime/View/AnimeViewController.swift

```diff
@@ -0,0 +1,144 @@
+//
+//  ViewController.swift
+//  Anime-App
+//
+//  Created by Tobi Adegoroye on 09/08/2023.
+//
+
+import UIKit
+
+class AnimeViewController: UIViewController {
+    
+    var searchTask: DispatchWorkItem?
+    private let animeViewModel = AnimeViewModel()
+    private var animes: [Anime] = [] {
+        didSet {
+            DispatchQueue.main.async {
+                self.tableView.reloadData()
+            }
+        }
+    }
+    
+    private var tableView: UITableView = {
+        let tableView = UITableView()
+        tableView.translatesAutoresizingMaskIntoConstraints = false
+        tableView.register(AnimeTableViewCell.self, forCellReuseIdentifier: AnimeTableViewCell.cellID)
+        return tableView
+    }()
+ 
+    private let searchVc = UISearchController(searchResultsController: nil)
+ 
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setup()
+    }
+}
+
+extension AnimeViewController: UISearchBarDelegate {
+    
+    func searchBar(_ searchBar: UISearchBar, textDidChange searchText: String) {
+        guard let searchText = searchBar.text?.lowercased(), !searchText.isEmpty else {
+            return
+        }
+        
+        guard let searchText = searchBar.text?.lowercased(), !searchText.isEmpty else {
+            return
+        }
+        self.searchTask?.cancel()
+
+        let task = DispatchWorkItem { [weak self] in
+            DispatchQueue.global(qos: .userInteractive).async { [weak self] in
+              //Use search text and perform the query
+        
+                self?.animeViewModel.searchAnime(query: searchText) { res in
+                    switch res {
+                    case .success(let anime):
+                        self?.animes = anime
+                        DispatchQueue.main.async {
+                            self?.tableView.reloadData()
+
+                        }
+                    case .failure(let error):
+                        print(error)
+                    }
+                }
+              DispatchQueue.main.async {
+                //Update UI
+              }
+            }
+          }
+          
+          self.searchTask = task
+          
+          //0.5 is the wait or idle time for execution of the function applyFilter
+          DispatchQueue.main.asyncAfter(deadline: DispatchTime.now() + 1, execute: task)
+        
+        
+    }
+    
+    func searchBarCancelButtonClicked(_ searchBar: UISearchBar) {
+        self.searchVc.dismiss(animated: true)
+    }
+    func searchBarSearchButtonClicked(_ searchBar: UISearchBar) {
+
```

Comment:
If you don't need this remove it don't be messy Tobi, stuff like this make a difference...